### PR TITLE
Attempt to fix icon theme fallbak

### LINKED ISSUE #330: The icons in Crow Translate lost even I have tried to configure the qt5ct
**Bug description**
The icons in Crow Translate lost even I have tried to configure the qt5ct. I don't know how to solve this problem.

**Steps to reproduce**
1. Install Crow Translate by dpkg
2. Install libqt5svg5, qt5ct by apt-get
3. add `export QT_QPA_PLATFORMTHEME=qt5ct` to ~/.profile
4. restart and run with `QT_QPA_PLATFORMTHEME=qt5ct crow`

**Screenshots**
image
image
image
image

**Test**
* If I set Appearence-style to gtk2 or run with `QT_QPA_PLATFORMTHEME=gtk2 crow`, "pk-gtk-module" error will occur. Some of the icons in the main interface lost, but the icons in Settings work well.
image
image
image
* If I use other styles, all icons almost lost.
image
image
image

**Environment**

- OS: Ubuntu 20.04.2 LTS
- Desktop environment: GNOME 3.36.8
- Crow Translate version: 2.8.4

**Additional context**
I have read https://github.com/crow-translate/crow-translate/issues/198 and googled a lot, but I still can't deal with trouble about QT. I need help.

### PR DESCRIPTION
**Pull request by gasinvein**
It seems like in order to fallbak to builtim icon theme from `.qrc`, the aliases format have to match the usual filesystem path format for icons, i.e. the top-level "directory" has to be `${size}x${size}` instead of just `${size}`.

Can fix #330 

Draft until confirmed not to break on other platforms/DEs.

### CODE REVIEW

**Comment by Shatur** [comment]
Oh, maybe it's because Gnome uses `index.theme` from system, not the built-in. Nice catch!
But it will be broken on Windows since it reads the built-in `index.theme` which uses [different format](https://github.com/vinceliuice/Fluent-icon-theme/blob/ea9510704e9f1117adeb92c75cf8bd771d75cd15/src/index.theme#L32).
But maybe we could just replace the icon theme which uses `${size}x${size}` in its format? Maybe you know a good theme with this constraint?

**Comment by gasinvein** [comment]
Does Windows use `index.theme` at all? I mean, icon themes aren't a thing in Windows.

**Comment by gasinvein** [comment]
Hmm. Since we're using only a small subset of the icon theme, maybe we can provide our own `index.theme` instead of the one shipped with the theme?

**Comment by Shatur** [comment]
This will work, but harder to maintain... Just use another theme should be easier?

**Comment by gasinvein** [comment]
I don't know much of icon packs. Oxygen icons maybe? It has `${size}x${size}` icon dirs, but it's mostly PNG.

**Comment by Shatur** [comment]
No, I think it should be something modern and SVG.

**Comment by gasinvein** [comment]
I've committed a minimal custom `index.theme` file. As it is, doesn't seem to add that much of maintenance burden.

**Comment by Shatur** [comment]
Looks nice. It works on GNOME, right? Will check on KDE and Windows today after work.

**Comment by gasinvein** [comment]
Yes, tested on GNOME.

**Comment by Shatur** [comment]
Works on KDE, but no icons on Windows. Maybe some information in this custom file is missing?

**Comment by gasinvein** [comment]
That's possible. Added some more information.

File: data/icons/fluent-icon-theme.index.theme [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+[16x16/status]
+Context=Status
+Size=16
+Type=Fixed
+
+[32x32/categories]
+Context=Applications
+Size=32
+Type=Fixed
```
Comment (Shatur) [inline data/icons/fluent-icon-theme.index.theme:25]: Checked, the issue was in folders. We can remove all other info, only `Size` is needed.

**Comment by Shatur** [comment]
And we also need to add `scalable category` for this icons:
https://github.com/crow-translate/crow-translate/blob/master/data/icons/freedesktop/freedesktop.qrc

**Comment by Shatur** [comment]
What if we unite this two files into a single one? We can do it thanks to migration to submodules.

File: data/icons/fluent-icon-theme.index.theme (hunk 1 above)
Comment (gasinvein) [reply data/icons/fluent-icon-theme.index.theme:25]: Not sure if I understood what you mean. Only e.g.
```
[16x16/status]
Size=16
```
is needed?

Comment (Shatur) [reply data/icons/fluent-icon-theme.index.theme:25]: Yep. For each category.

Comment (gasinvein) [reply data/icons/fluent-icon-theme.index.theme:25]: Hmm, without the `Context` property, declaring 2nd-level directories seems pointless. Are you sure `Context` isn't needed?

**Comment by gasinvein** [comment]
> What if we unite this two files into a single one? 

That should be doable, although I personally think 3rd-party icons should be kept separate, as any other 3rd-party stuff.

**Comment by Shatur** [comment]
> That should be doable, although I personally think 3rd-party icons should be kept separate, as any other 3rd-party stuff.

Agree, but very easy to forget to update since their paths should match.

File: data/icons/fluent-icon-theme.index.theme (hunk 1 above)
Comment (Shatur) [reply data/icons/fluent-icon-theme.index.theme:25]: Yep.

Comment (Shatur) [reply data/icons/fluent-icon-theme.index.theme:25]: This is how it was you pushed the first time, but with correct folder names.

Comment (gasinvein) [reply data/icons/fluent-icon-theme.index.theme]: Oh well, removed all properties except `Size`

**Comment by gasinvein** [comment]
> > That should be doable, although I personally think 3rd-party icons should be kept separate, as any other 3rd-party stuff.
> 
> Agree, but very easy to forget to update since their paths should match. Since we provide our `fluent-icon-theme.index.theme` it should be fine. I would also rename it to just `fluent-icon-theme.theme`.

Ok. But since it's not really Fluent icon theme anymore, maybe rename those files to `icon-theme.qrc` and `icon-theme.theme`?

**Comment by Shatur** [comment]
> Ok. But since it's not really Fluent icon theme anymore, maybe rename those files to icon-theme.qrc and icon-theme.theme?

Agree!

**Comment by Shatur** [comment]
Maybe even `index.theme`?

**Comment by gasinvein** [comment]
Done. I hope I didn't mess anything up.

**Comment by Shatur** [comment]
Now some icons are blurry... I think we need to put `Type=Fixed` back.

**Comment by Shatur** [comment]
No, didn't helped. Hm...

**Comment by Shatur** [comment]
Never mind, the same behavior is on master. Will fix it in separate. Thank you a lot for the PR!

**Pull request merged**